src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -2211,6 +2212,18 @@ BlockIO InterpreterCreateQuery::doCreateOrReplaceTable(ASTCreateQuery & create,
 
 BlockIO InterpreterCreateQuery::fillTableIfNeeded(const ASTCreateQuery & create)
 {
+    /// If the query is a CREATE AS SELECT, we need to run select query as INITIAL_QUERY
+    /// to ensure that the query kind is set correctly for the select query.
+    auto previous_query_kind = getContext()->getClientInfo().query_kind;
+    SCOPE_EXIT({
+        getContext()->setQueryKind(previous_query_kind);
+    });
+    getContext()->setQueryKind(ClientInfo::QueryKind::INITIAL_QUERY);
```

Comment:
Please check `CREATE AS SELECT` queries (w/o REPLACE). AFAIS, it's different code path and doesn't use `Context::setQueryContext()`

Reply:
I do not understand what do you mean.

`InterpreterCreateQuery::doCreateOrReplaceTable` also calls `fillTableIfNeeded`.